cdap-app-templates/cdap-program-report/cdap-program-report-base/src/main/java/co/cask/cdap/report/ReportGenerationSpark.java

```diff
@@ -176,43 +228,103 @@ private ReportList getReportList(int offset, int limit) throws Exception {
       return new ReportList(offset, limit, reportIdDirs.size(), reportStatuses);
     }
 
+    @POST
+    @Path("/reports/{report-id}/share")
+    public void shareReport(HttpServiceRequest request, HttpServiceResponder responder,
+                            @PathParam("report-id") String reportId) {
+      Location reportIdDir;
+      try {
+        String userName = getUserName(request.getAllHeaders());
+        reportIdDir = getDatasetBaseLocation(ReportGenerationApp.REPORT_FILESET).append(userName).append(reportId);
+        if (!reportIdDir.exists()) {
+          responder.sendError(500, String.format("Invalid report-id %s, report does not exist", reportId));
```

Comment:
why it is 500? It is a user error, right? Shouldn't it be 4xx?

Reply:
yes, updated it to 404.